File: CSharp/Tests/AddTwoNumbers.cs
using Solutions;

namespace Tests;

public class AddTwoNumbersTest {
    [Fact]
    public void AddTwoNumbers_Case1() {
        ListNode l1 = new(2, new ListNode(4, new ListNode(3)));
        ListNode l2 = new(5, new ListNode(6, new ListNode(4)));
        ListNode want = new(7, new ListNode(0, new ListNode(8)));

        ListNode result = AddTwoNumbers.AddTwoNumbers1(l1, l2);
        Assert.Equivalent(want, result);
    }

    [Fact]
    public void AddTwoNumbers_Case2() {
        ListNode l1 = new(0);
        ListNode l2 = new(0);
        ListNode want = new(0);

        ListNode result = AddTwoNumbers.AddTwoNumbers1(l1, l2);
        Assert.Equivalent(want, result);
    }

    [Fact]
    public void AddTwoNumbers_Case3() {
        ListNode l1 = new(
            9, new ListNode(
                9, new ListNode(
                    9, new ListNode(
                        9, new ListNode(
                            9, new ListNode(
                                9, new ListNode(9)))))));
        ListNode l2 = new(
            9, new ListNode(
                9, new ListNode(
                    9, new ListNode(9))));
        ListNode want = new(
            8, new ListNode(
                9, new ListNode(
                    9, new ListNode(
                        9, new ListNode(
                            0, new ListNode(
                                0, new ListNode(
                                    0, new ListNode(1))))))));

        ListNode result = AddTwoNumbers.AddTwoNumbers1(l1, l2);
        Assert.Equivalent(want, result);
    }
}

File: Rust/tests/add_two_numbers.rs
#[cfg(test)]
mod tests {
    use leetcode::add_two_numbers::*;
    use leetcode::create_linked_list;

    #[test]
    fn test_add_two_numbers_case1() {
        let l1 = create_linked_list(&[2, 4, 3]);
        let l2 = create_linked_list(&[5, 6, 4]);
        let want = create_linked_list(&[7, 0, 8]);
        let result = add_two_numbers(l1, l2);
        assert_eq!(result, want);
    }

    #[test]
    fn test_add_two_numbers_case2() {
        let l1 = create_linked_list(&[0]);
        let l2 = create_linked_list(&[0]);
        let want = create_linked_list(&[0]);
        let result = add_two_numbers(l1, l2);
        assert_eq!(result, want);
    }

    #[test]
    fn test_add_two_numbers_case3() {
        let l1 = create_linked_list(&[9, 9, 9, 9, 9, 9, 9]);
        let l2 = create_linked_list(&[9, 9, 9, 9]);
        let want = create_linked_list(&[8, 9, 9, 9, 0, 0, 0, 1]);
        let result = add_two_numbers(l1, l2);
        assert_eq!(result, want);
    }
}
